designdocs/Capacity.md

```diff
@@ -0,0 +1,657 @@
+
+Similarly, we impose a limit on how much space Capacity transactions can consume from Normal transactions. This new configurable limit can be set by governance.
+
+![https://user-images.githubusercontent.com/3433442/189949020-7bdd2e34-5323-4264-a821-1dcbb0063c20.png](https://user-images.githubusercontent.com/3433442/189949020-7bdd2e34-5323-4264-a821-1dcbb0063c20.png)
+
+A [SignedExtension](https://paritytech.github.io/substrate/master/sp_runtime/traits/trait.SignedExtension.html) trait is implemented so that once the Capacity transaction has reached the `max_total` of allocated Capacity space, the transaction is put back into the transaction pool. Below illustrates the Capacity transaction SignedExtension flow. 
+
+![https://user-images.githubusercontent.com/3433442/189949048-7d19a194-701d-4267-ae1a-0333ee665ae7.png](https://user-images.githubusercontent.com/3433442/189949048-7d19a194-701d-4267-ae1a-0333ee665ae7.png)
+
+A type for implementing the SignedExtension trait:
+
+```rust
+/// A type that implements SignedExtension that checks to see if Capacity transaction allocated
+/// weight has been reached.
+pub struct CheckCapacityWeight<T: Config + Send + Sync>(sp_std::market::PhantomData<T>);
+```
+
+```rust
+impl<T: Config + Send + Sync> SignedExtension for CheckCapacityWeight<T>
+	where T::Call: Dispachable<Info = DispatchInfo> + IsSubtype<Call<T>>,
+{
+	type AccountId = T::AccountId;
+	type Call = T::Call;
+	type AdditionalSigned = ();
+	type Pre = ();
+
+	fn additional_signed(&self) -> sp_std::result::Result<(), TransactionValidityError> {
+		Ok(())
+	}
+
+	fn pre_validate() -> Result<(), TransactionValidityError> {}
+  
+  /// Below describes the interfaces for validate, pres_dispatch and post_dispatch
+}
+```
+
+SignedExtension validate
+
+```rust
+/// Validates that extrinsic does not exceed max-total of Capacity transactions
+///
+/// ### Errors
+///
+/// - Returns InvalidTransaction::ExhaustsResource if transaction is greater than
+///   max-total for Capacity Transactions
+fn validate(
+	&self,
+	_who: &Self::AccountId,
+	call: &Self::Call,
+	info: &DispatchInfoOf<Self::Call>,
+  len: usize,
+) -> TransactionValidity {}
+```
+
+Acceptance Criteria are listed below but can evolve:
+
+1. Checks that the extrinsic does not exceed the size of the `max_total` allocated space.
+
+SignedExtension pre-dispatch
+
+```rust
+/// Validates that extrinsic does not exceed max-total of Capacity transactions
+///
+/// ### Errors
+///
+/// - Returns InvalidTransaction::ExhaustsResource if transaction fails checks.
+fn pre_dispatch(
+		self,
+    _who: &Self::AccountId,
+    _call: &Self::Call,
+    info: &DispatchInfoOf<Self::Call>,
+    len: usize,
+	) -> Result<(), TransactionValidityError> {}
+```
+
+Acceptance Criteria are listed below but can evolve:
+
+1. Only run validation, pre-dispatch, and post-dispatch on calls that match Capacity Transactions.
```

Comment:
is it possible to separate this list into 2 categories? The old checks and new capacity related ones